doc/releases/release-notes-3.7.rst

```diff
@@ -214,6 +214,8 @@ Boards & SoC Support
   * Added support for Ambiq Apollo3 boards: ``apollo3_evb``, ``apollo3p_evb``.
   * Added support for RaspberryPi5 board: ``rpi5``.
   * Added support for Seeed Studio XIAO RP2040 board: ``xiao_rp2040``.
+  * Added support for Mikroe RA4M1 Clicker board: ``mikroe_clicker_ra4m1``.
```

Comment:
A suggestion, given that this board has a dedicated page it may be nicer to have this be a link to that page(?)

Reply:
100% agreed, and not the first time I'm thinking we should do it. This does apply to all the boards though, not just the one in this PR. Maybe worth keeping as-is and taking care of adding cross-links in the release notes closer to the release date, as part of putting the final touches to it? I can volunteer myself to help with that :)

Reply:
I attempted to refine it.
Could you review it?